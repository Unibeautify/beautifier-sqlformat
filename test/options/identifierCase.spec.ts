import * as fs from "fs";
import * as path from "path";

import { newUnibeautify, Beautifier } from "unibeautify";
import beautifier from "../../src";
import { raw } from "../utils";

// tslint:disable:mocha-no-side-effect-code
test(`should uppercase SQL identifiers`, () => {
  const text: string = fs
    .readFileSync(path.resolve(__dirname, `../fixtures/test2.sql`))
    .toString();
  const unibeautify = newUnibeautify();
  unibeautify.loadBeautifier(beautifier);
  return unibeautify
    .beautify({
      languageName: "SQL",
      options: {
        SQL: {
          identifier_case: "uppercase",
        },
      },
      text,
    })
    .then(results => {
      expect(raw(results)).toMatchSnapshot();
    });
});
test(`should capitalize SQL identifiers`, () => {
  const text: string = fs
    .readFileSync(path.resolve(__dirname, `../fixtures/test2.sql`))
    .toString();
  const unibeautify = newUnibeautify();
  unibeautify.loadBeautifier(beautifier);
  return unibeautify
    .beautify({
      languageName: "SQL",
      options: {
        SQL: {
          identifier_case: "capitalize",
        },
      },
      text,
    })
    .then(results => {
      expect(raw(results)).toMatchSnapshot();
    });
});
test(`should lowercase SQL identifiers`, () => {
  const text: string = fs
    .readFileSync(path.resolve(__dirname, `../fixtures/test2.sql`))
    .toString();
  const unibeautify = newUnibeautify();
  unibeautify.loadBeautifier(beautifier);
  return unibeautify
    .beautify({
      languageName: "SQL",
      options: {
        SQL: {
          identifier_case: "lowercase",
        },
      },
      text,
    })
    .then(results => {
      expect(raw(results)).toMatchSnapshot();
    });
});
