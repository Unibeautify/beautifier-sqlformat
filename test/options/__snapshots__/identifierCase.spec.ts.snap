// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should capitalize SQL identifiers 1`] = `
Object {
  Symbol(raw): "select *
FROM Users U
join Users_permission Up on U.User_id = Up.User_id
where U.User_id = 1
  and Up.Permission_id = 2;",
}
`;

exports[`should lowercase SQL identifiers 1`] = `
Object {
  Symbol(raw): "select *
FROM users u
join users_permission up on u.user_id = up.user_id
where u.user_id = 1
  and up.permission_id = 2;",
}
`;

exports[`should uppercase SQL identifiers 1`] = `
Object {
  Symbol(raw): "select *
FROM USERS U
join USERS_PERMISSION UP on U.USER_ID = UP.USER_ID
where U.USER_ID = 1
  and UP.PERMISSION_ID = 2;",
}
`;
